completion/_catkin

```diff
@@ -102,7 +102,8 @@ _catkin_packages_caching_policy() {
   if [[ -z "$_workspace_source_space" ]]; then
     _debug_log "Searching for source space..."
 
-    _workspace_source_space="$(catkin locate -s --quiet)"
+    _catkin_get_enclosing_workspace $_workspace_root_hint
+    _workspace_source_space="$_workspace_root/src"
```

Comment:
This is a flawed assumption, the source directory is not always `<workspace dir>/src`. Since it can be any folder, that's why `catkin locate` is required to look up the setting.